src/server/db_slice.h

```diff
@@ -497,6 +497,17 @@ class DbSlice {
   void PerformDeletion(Iterator del_it, DbTable* table);
   void PerformDeletion(PrimeIterator del_it, DbTable* table);
 
+  ThreadLocalMutex* GetThreadLocalMutex() {
```

Comment:
I'd return a reference instead of a pointer. It reduces edge case handling / assumptions on both caller and callee

Reply:
Also, maybe call it `GetSerializationMutex()` or `GetIterationMutex()`?
I'd also add a comment saying what this guards and when it is necessary to use it

Reply:
> I'd return a reference instead of a pointer. It reduces edge case handling / assumptions on both caller and callee

This breaks the style guide.

Reply:
The styleguide never forbid returning references, even mutable ones. Only accepting parameters as mutable references. That's no longer the case as you remember, but unfortunately we continue to follow that.
But returning a mutable reference has always been ok...